rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -0,0 +1,311 @@
+promoted as a user-facing feature. Instead, the guidance since then has been 
+to use the task_group or the flow graph APIs to express patterns that were 
+previously expressed using with the lowest level tasking API. And for most 
+cases, this has been sufficient. However, there are some use cases that create 
+dynamically expanding graphs of dependent tasks are awkward to express using 
+the existing interfaces and so this proposal expands tbb::task_group to make 
+additional use cases easier to express.
+
+The current class definition in the latest oneTBB specification for 
+`tbb::task_group` is shown below. Note the `defer` function because this 
+function and its return type, `task_handle`, are the foundation of our proposed
+extensions:
+
+    class task_group {
+    public:
+        task_group();
+        task_group(task_group_context& context);
+
+        ~task_group();
+
+        template<typename Func>
+        void run(Func&& f);
+
+        template<typename Func>
+        task_handle defer(Func&& f);
+
+        void run(task_handle&& h);
+
+        template<typename Func>
+        task_group_status run_and_wait(const Func& f);
+
+        task_group_status run_and_wait(task_handle&& h);
+
+        task_group_status wait();
+        void cancel();
+    };
+
+## Proposal
+
+The following table summarizes the three primary extensions that are under 
+consideration. The remainder of this post provides background and further 
+clarification on these proposed extensions.
+
+1. Extend semantics and useful lifetime of `task_handle`. We propose `task_handle` to represent tasks for the purpose of adding dependencies. The useful lifetime and semantics of `task_handle` will need to be extended to include tasks that have been submitted, are currently executing, or have been completed.
+2. Add functions to set task dependencies. In the current `task_group`, tasks can only be waited for as a group and there is no direct way to add any before-after relationships between individual tasks. We will discuss options for spelling.
+3. Add a function to move successors from a currently executing task to a new task. This functionality is necessary for recursively generated task graphs. This case represents a situation where it is safe to modify dependencies for an already submitted task. 
+
+### Extend the semantics and useful lifetime of task_handle
+
+Dynamic tasks graphs order the execution of tasks via dependencies on the 
+completion of other tasks. They are dynamic because the creation of tasks, 
+specification of dependencies between, submission of tasks for scheduling, and
+execution of the tasks may happen concurrently and in various orders. Different
+concrete use cases have different requirements on when new tasks are created 
+and when dependencies between tasks are specified. 
+
+For the sake of discussion, let’s label four points in a task’s lifetime: 
+
+1. pre-submitted
```

Comment:
Would it make sense to call this deferred?

Reply:
Alternatively, call it "created" in line with the text below

Reply:
I like "deferred" since it matches the defer function name. I think it will accurately imply that it is not yet submitted. I had been concerned that a submitted task that has not-yet-ready-to-run because of dependencies could also be thought of as "deferred", but on second thought, I think deferred is different enough.

Reply:
I suggest that we are careful with the term "deferred". A deferred task has a specific meaning in OpenMP. While OpenMP is not the "bible", I think it is clever to stick to their semantics or to explicitly avoid their terms where we disagree with their usage. So in this sense, my idea with "deferred" was wrong. I should have thought about this more carefully.

Reply:
Already changed it. Do you have a better suggestion instead? So maybe "created" then?